retro/reco.py

```diff
@@ -103,196 +116,361 @@ def run(self, method):
+                self.setup_hypo(
+                                cascade_kernel='scaling_aligned_one_dim',
+                                track_kernel='pegleg',
+                                track_time_step=1.,
+                                )
 
-            if method == 'truth':
-                settings = self.run_with_truth(
-                    prior=prior,
-                    loglike=loglike,
-                    truth=event['truth'],
-                )
+                self.hypo_handler.fixed_params = OrderedDict()
+
+                return_param_values = []
+                return_log_likelihoods = []
 
-            elif method == 'mymini':
-                settings = self.run_mymini(
-                    prior=prior,
-                    loglike=loglike,
+                # Setup prior
+                prior_defs = OrderedDict()
+                
+                est_x = estimate['weighted_mean']['x']
+                est_y = estimate['weighted_mean']['y']
+                est_z = estimate['weighted_mean']['z']
+                est_time = estimate['weighted_mean']['time']
+
+                prior_defs['x'] = {'kind':'cauchy', 'loc':est_x, 'scale':15, 'low':est_x-300, 'high':est_x+300}
+                prior_defs['y'] = {'kind':'cauchy', 'loc':est_y, 'scale':15, 'low':est_y-300, 'high':est_y+300}
+                prior_defs['z'] = {'kind':'cauchy', 'loc':est_z, 'scale':10, 'low':est_z-200, 'high':est_z+200}
+                prior_defs['time'] = {'kind':'cauchy', 'loc':est_time, 'scale':40, 'low':est_time-800, 'high':est_time+800}
+                self.generate_prior(prior_defs)
+
+
+                # Setup llh function
+                self.generate_loglike(
+                    return_param_values=return_param_values,
+                    return_log_likelihoods=return_log_likelihoods,
+                    t_start=t_start,
                 )
 
-            elif method == 'multinest':
                 settings = self.run_multinest(
-                    prior=prior,
-                    loglike=loglike,
-                    **self.reco_kw
+                    importance_sampling=True,
+                    max_modes=1,
+                    const_eff=True,
+                    n_live=250,
+                    evidence_tol=0.1,
+                    sampling_eff=0.3,
+                    max_iter=10000,
+                    seed=0,
                 )
-            elif method == 'scipy':
-                settings = self.run_scipy(
-                    prior=prior,
-                    loglike=loglike,
-                    method='differential_evolution',
-                    eps=0.02
+
+                t2 = time.time()
+
+                llhp = self.make_llhp(return_log_likelihoods, return_param_values, fname=None)
+                opt_meta = self.make_meta_dict(settings, llhp=llhp, time=t2-t1, fname='opt_meta')
+                estimate = self.make_estimate(llhp, opt_meta, fname='estimate')
+
+                # -------- 10D ---------
```

Comment:
the 10D stuff is commented out; can this either be enabled as a different method or via kwarg or global MULTINEST_USE_10D type variable or somesuch so we can turn it on / off without commenting / uncommenting a bunch of code? (if this code doesn't work and is just here to help you/me implement this "correctly" at some point, then feel free to leave it in though, just add a comment above to that effect)

Reply:
it does work, but didn't deliver any good resolutions....these recipes are all still experimental, that's why I don't really want to created 20 different ones...